talk/owt/sdk/p2p/p2pclient.cc

```diff
@@ -251,79 +256,88 @@ void P2PClient::OnSignalingMessage(const std::string& message,
-      // If we don't have a PC before we receive an offer.
-      auto pcc = that->GetPeerConnectionChannel(remote_id);
-      if (pcc->HaveLocalOffer() && that->local_id_.compare(remote_id) > 0) {
-        // If our ID is larger than remote, make the remote side as the
-        // publisher (offerer) In case we already have an offer. So we
-        // remove current PCC and create answer.
-        std::shared_ptr<LocalStream> stream = pcc->GetLatestLocalStream();
-        std::function<void()> success_callback =
-            pcc->GetLatestPublishSuccessCallback();
-        std::function<void(std::unique_ptr<Exception>)> failure_callback =
-            pcc->GetLatestPublishFailureCallback();
-        pcc->Stop(nullptr, nullptr);
-        {
-          // If we already created offer,
-          const std::lock_guard<std::mutex> lock(that->pc_channels_mutex_);
-          that->pc_channels_.erase(remote_id);
-        }
-        auto new_pcc = that->GetPeerConnectionChannel(remote_id);
-        new_pcc->OnIncomingSignalingMessage(message);
-        new_pcc->Publish(stream, success_callback, failure_callback);
-        return;
-      }
-    } else if (message.find("\"type\":\"chat-closed\"") != std::string::npos) {
-      RTC_LOG(LS_ERROR) << "Handle the situation that PCC is created by "
-                           "received Chat-closed.";
-      int code = 0;
-      std::string error = "";
-      Json::Reader reader;
-      Json::Value json_message;
-      if (reader.parse(message, json_message)) {
-        Json::Value stop_info;
-        rtc::GetValueFromJsonObject(json_message, "data", &stop_info);
-        rtc::GetIntFromJsonObject(stop_info, "code", &code);
-        rtc::GetStringFromJsonObject(stop_info, "message", &error);
-
-        if (code == kWebrtcIceGatheringPolicyUnsupported) {
-          auto pcc = that->GetPeerConnectionChannel(remote_id);
+    } else if (message_type == "chat-signal"){
+      Json::Value signal;
+      rtc::GetValueFromJsonObject(json_message, kMessageDataKey, &signal);
+      std::string sdp_type;
+      rtc::GetStringFromJsonObject(signal, kSessionDescriptionTypeKey, &sdp_type);
+      if (sdp_type == "offer") {
+        // If we don't have a PC before we receive an offer.
+        auto pcc = that->GetPeerConnectionChannel(remote_id);
+        if (pcc->HaveLocalOffer() && that->local_id_.compare(remote_id) > 0) {
+          // If our ID is larger than remote, make the remote side as the
+          // publisher (offerer) In case we already have an offer. So we
+          // remove current PCC and create answer.
           std::shared_ptr<LocalStream> stream = pcc->GetLatestLocalStream();
           std::function<void()> success_callback =
               pcc->GetLatestPublishSuccessCallback();
           std::function<void(std::unique_ptr<Exception>)> failure_callback =
               pcc->GetLatestPublishFailureCallback();
-          pcc->SetAbandoned();
+          pcc->Stop(nullptr, nullptr);
```

Comment:
The logical for "offer" is unchanged. Old version (line 272) uses pcc->Stop as well. Not sure why it's identified as a change.

Reply:
Adding @dvrogozh